src/main/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidator.java

```diff
@@ -183,6 +209,141 @@ private void validateInconsistentPeriodFiling(Debtors debtors, Errors errors) {
             addInconsistentDataError(errors, PREVIOUS_TOTAL_PATH);
         }
     }
-}
 
+    /**
+     * @inheritDoc
+     */
+
+    @Override
+    public Errors crossValidate(Errors errors, HttpServletRequest request, String CompanyAccountsId,
+                                Debtors debtors) throws DataException {
+
+        crossValidateCurrentPeriod(errors, request, debtors, CompanyAccountsId);
+        crossValidatePreviousPeriod(errors, request, CompanyAccountsId, debtors);
+
+        return errors;
+    }
+
+    private void crossValidatePreviousPeriod(Errors errors, HttpServletRequest request,
+                                             String CompanyAccountsId,
+                                             Debtors debtors) throws DataException {
+
+        String previousPeriodId = previousPeriodService.generateID(CompanyAccountsId);
+
+        // get previous period object
+        ResponseObject<PreviousPeriod> previousPeriodResponseObject;
+
+        try {
+            previousPeriodResponseObject =
+                previousPeriodService.findById(previousPeriodId, request);
+        } catch (MongoException e) {
+
+            throw new DataException(e.getMessage(), e);
+        }
+
+        // if note is null and balance sheet value isn't add error
+        if (isDebtorsNotePreviousTotalNull(debtors) &&
```

Comment:
We could move this to a different method:  `checkIfNoteIsNullAndBalanceNot()`
So that you could describe what this method is doing instead of using the comments.